lib/interactor.rb

```diff
@@ -163,4 +163,30 @@ def call
   # Returns nothing.
   def rollback
   end
+
+  private
+
+  # Internal: Determine what arguments (if any) should be passed to the "call"
+  # instance method when invoking an Interactor. The "call" instance method may
+  # accept any combination of positional and keyword arguments. This method
+  # will extract values from the context in order to populate those arguments
+  # based on their names.
+  #
+  # Returns an Array of arguments to be applied as an argument list.
+  def arguments_for_call # rubocop:disable Metrics/MethodLength
```

Comment:
The only thing that bothers me is that supporting only keywords arguments may be confusing. `#call` will not act like a standard Ruby method, but will have its own rules instead. There would be additional lines in Readme, things to keep in memory, things to explain to a newcomer into a project-which-use-interactors, etc.

I thought about alternative syntax:

```ruby
class MyInteractor
  include Interactor

  required_arguments :user, :order
  optional_arguments :email, send_notification: true

  def call
  end
end
```

On the one hand, keyword arguments do the same and in more expressive way and they are already familiar.
On the other hand, this familiarity may become a bad thing in our case. E.g, if I can write:

```ruby
def call(hello:)
end

MyInteractor.call(hello: "Steve")
```

why couldn't I write

```ruby
def call(name)
end
MyInteractor.call("Steve")
```

? And here magic falls apart.